eval/unwrap.go

```diff
@@ -97,12 +97,37 @@ func (u ValueUnwrapper) NonNegativeInt() (int, error) {
 	return i, u.err
 }
 
+func (u ValueUnwrapper) Fd() (int, error) {
+	s, err := u.String()
+	if err != nil {
+		return 0, err
+	}
+	switch s {
+	case "stdin":
+		return 0, nil
+	case "stdout":
+		return 1, nil
+	case "stderr":
+		return 2, nil
+	default:
+		i, errInt := u.NonNegativeInt()
+		if errInt != nil {
+			return 0, fmt.Errorf("Fd must be standard stream names or integer; got %s", s)
```

Comment:
Can you use `u.errorpf` to build the error, like in the `error` method above? This way the error will have the correct information about the range.